docs/source/python/data.rst

```diff
@@ -560,3 +560,59 @@ schema without having to get any of the batches.::
    x: int64
 
 It can also be sent between languages using the :ref:`C stream interface <c-stream-interface>`.
+
+Conversion of RecordBatch do Tensor
+-----------------------------------
+
+Each array of the ``RecordBatch`` has it's own contiguous memory that is not necessarily
+adjacent to other arrays. A different memory structure that is used in machine learning
+libraries is a two dimensional array (also called a 2-dim tensor or a matrix) which takes
+only one contiguous block of memory.
+
+For this reason there is a function ``pyarrow.RecordBatch.to_tensor()`` available
+to efficiently convert tabular columnar data into a matrix.
+
+Data types supported in this conversion are unsigned, signed integer and float
+types of all widths. Currently only column-major conversion is supported.
```

Comment:
I think it is sufficient to say that it supports integer and float data types, I think when written generally like that it is clear this is for both signed and unsigned of all widths